sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -202,40 +200,15 @@ public Symbol generateQuerySymbol(Optional<Expression> whereExpression, Expressi
         }
     }
 
-    private FunctionInfo getBuiltinFunctionInfo(String name, List<DataType> argumentTypes) {
-        FunctionImplementation impl = functions.getBuiltin(name, argumentTypes);
-        if (impl == null) {
-            throw Functions.createUnknownFunctionException(name, argumentTypes);
-        }
-        return impl.info();
-    }
-
-    private FunctionInfo getBuiltinOrUdfFunctionInfo(@Nullable String schema, String name, List<DataType> argumentTypes) {
-        FunctionImplementation impl;
-        if (schema == null) {
-            impl = functions.getBuiltin(name, argumentTypes);
-            if (impl == null) {
-                SessionContext sessionContext = transactionContext.sessionContext();
-                impl = functions.getUserDefined(sessionContext.defaultSchema(), name, argumentTypes);
-            }
-        } else {
-            impl = functions.getUserDefined(schema, name, argumentTypes);
-        }
-        return impl.info();
-    }
-
     protected Symbol convertFunctionCall(FunctionCall node, ExpressionAnalysisContext context) {
         List<Symbol> arguments = new ArrayList<>(node.getArguments().size());
-        List<DataType> argumentTypes = new ArrayList<>(node.getArguments().size());
         for (Expression expression : node.getArguments()) {
             Symbol argSymbol = expression.accept(innerAnalyzer, context);
-
-            argumentTypes.add(argSymbol.valueType());
             arguments.add(argSymbol);
         }
 
         List<String> parts = node.getName().getParts();
-        String schema = null;
+        String schema = transactionContext.sessionContext().defaultSchema();
```

Comment:
Does this still have the same semantics if we always set the schema from the session? Afaik we wanted to treat `x()` with session schema `foo` differently than `foo.x()` (due to built-ins not being in any schema)

Reply:
Can we extract this? If it's a bugfix we also need to backport it